src/promptflow/promptflow/executor/_line_execution_process_pool.py

```diff
@@ -207,66 +324,171 @@ def __exit__(self, exc_type, exc_val, exc_tb):
             self._pool.close()
             self._pool.join()
 
-    def _timeout_process_wrapper(self, run_start_time: datetime, task_queue: Queue, timeout_time, result_list):
-        healthy_ensured_process = HealthyEnsuredProcess(self._executor_creation_func, self.context)
-        healthy_ensured_process.start_new(task_queue)
-
-        if not healthy_ensured_process.process.is_alive():
-            return
+    def _process_task(
+        self,
+        process_info,
+        run_start_time,
+        task_queue,
+        timeout_time,
+        result_list
+    ):
 
         while True:
+            restart_outer_loop = False
+            index, pid, process_name, input_queue, output_queue = process_info
+            # if not using fork, process_info's first element is healthy_ensured_process.
+            if not self._use_fork:
+                healthy_ensured_process = index
+
             try:
                 args = task_queue.get(timeout=1)
             except queue.Empty:
-                healthy_ensured_process.end()
-                return
+                if self._use_fork:
+                    self._control_signal_queue.put((pid, index, "del"))
+                    #  To prevent BrokenPipeError when process attempts to get data from the closed
+                    #  process, make sure the process have terminated before return
+                    while True:
+                        if not psutil.pid_exists(pid):
+                            return
+                        time.sleep(1)
+                else:
+                    healthy_ensured_process.end()
+                    return
 
-            healthy_ensured_process.put(args)
+            input_queue.put(args)
             inputs, line_number, run_id = args[:3]
-            self._processing_idx[line_number] = healthy_ensured_process.format_current_process(line_number)
 
+            self._processing_idx[line_number] = format_current_process(process_name, pid, line_number)
             start_time = datetime.utcnow()
             completed = False
 
             while datetime.utcnow().timestamp() - start_time.timestamp() <= timeout_time:
                 try:
+                    # Monitor process aliveness and start new one if it crashes
+                    if self._use_fork:
+                        if not psutil.pid_exists(pid):
+                            # If the process crashes, set the 'restart_outer_loop' to TRUE.
+                            # And re-execute the task from the beginning.
+                            restart_outer_loop = True
+                            # Clear the contents of input_queue to allow its reuse in fork mode.
+                            input_queue.get()
+                            # Put unfinished tasks into task_queue again.
+                            task_queue.put(args)
```

Comment:
I think we don't need this put back logic.
If the process crashes, we just handle it as failed.

You have already put the `args` into input_queue in line 358, if the process crashes after it get task from input queue, then `input_queue.get()` will stuck.

Reply:
When the process crashed, set this line to failed